src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -6372,6 +6419,14 @@ namespace SurfaceGeometry {
         Real64 SchedMinValue;
         Real64 SchedMaxValue;
 
+        if (state.dataSolarShading->GetInputFlag) {
+            SolarShading::GetShadowingInput(state);
+            state.dataSolarShading->GetInputFlag = false;
+            state.dataSolarShading->MaxHCV =
+                (((max(15, state.dataSurface->MaxVerticesPerSurface) + 16) / 16) * 16) - 1; // Assure MaxHCV+1 is multiple of 16 for 128 B alignment
+            assert((state.dataSolarShading->MaxHCV + 1) % 16 == 0);
+        }
+
```

Comment:
Same logic above for the detached shading surfaces. The shadow calculation methods input needs to be process first before executing logic for the warning. This will make sure the shadow calculation inputs has been processed first and the state variable will have updated value to ensure the logic outcome is determined using correct values.

Reply:
After last change and comments: Except now that it is calling a stripped-down version of input processing other than the full-blown one.